Shapes/rml_rules_shape.ttl

```diff
@@ -0,0 +1,1683 @@
+            ns1:nodeKind ns1:Literal ;
+            ns1:path <http://www.w3.org/ns/r2rml#template> ],
+        [ ns1:datatype xsd:string ;
+            ns1:description """
+        Name of a column in the logical table. When generating RDF triples from
+        a logical table row, value from the specified column is used as the 
+        subject, predicate, or object (based upon the specific domain).
+        """ ;
+            ns1:message """
+        rr:column must be a string.
+        """ ;
+            ns1:name "rr:column" ;
+            ns1:nodeKind ns1:Literal ;
+            ns1:path <http://www.w3.org/ns/r2rml#column> ],
+        [ ns1:description """
+        A property for indicating whether a term map is a constant-valued term 
+        map.
+        """ ;
+            ns1:message """
+        rr:constant must be an IRI.
+        """ ;
+            ns1:name "rr:constant" ;
+            ns1:nodeKind ns1:IRI ;
+            ns1:path <http://www.w3.org/ns/r2rml#constant> ],
+        [ ns1:datatype xsd:string ;
+            ns1:description """
+        A reference rml:reference is used to refer to a column in case of 
+        databases, a record in case of CSV or TSV data source, an element in 
+        case of XML data source, an object in case of a JSON data source, etc.
+
+        A reference must be a valid identifier, considering the reference 
+        formulation (rml:referenceFormulation) specified. The reference can be
+        an absolute path, or a path relative to the iterator specified at the 
+        logical source. 
+        """ ;
+            ns1:message """
+        rml:reference must be a string.
+        """ ;
+            ns1:name "rml:reference" ;
+            ns1:nodeKind ns1:Literal ;
+            ns1:path <http://semweb.mmlab.be/ns/rml#reference> ] .
+
+<http://schema.org/RRobjectShape> a ns1:NodeShape ;
+    ns1:closed true ;
+    ns1:description """
+    An IRI reference for use as object for all the RDF triples generated from 
+    a logical table row or iterator.
+    """ ;
+    ns1:ignoredProperties ( rdf:type ) ;
+    ns1:message """
+    rr:object shortcut for rr:objectMap must be an IRI.
+    """ ;
+    ns1:name "rr:object" ;
+    ns1:nodeKind ns1:BlankNodeOrIRI .
+
+<http://schema.org/RRpredicateShape> a ns1:NodeShape ;
+    ns1:closed true ;
+    ns1:description """
+    Specifies the predicate for the generated triple from the logical table 
+    row or iterator.
+    """ ;
+    ns1:ignoredProperties ( rdf:type ) ;
+    ns1:message """
+    rr:predicate must be an IRI.
+    """ ;
+    ns1:name "rr:predicate" ;
+    ns1:nodeKind ns1:IRI .
+
+<http://schema.org/RRsubjectShape> a ns1:NodeShape ;
+    ns1:closed true ;
+    ns1:description """
+    An IRI reference for use as subject for all the RDF triples generated from 
+    a logical table row or iterator.
+    """ ;
+    ns1:ignoredProperties ( rdf:type ) ;
+    ns1:message """
+    rr:subject shortcut for rr:subjectMap must be an IRI.
```

Comment:
Can be either IRI or blank node

Reply:
True, that's a typo.
Also needs a change a bit further with `nodeKind`